pkg/ingester/ingester.go

```diff
@@ -529,6 +572,9 @@ func (i *Ingester) Collect(ch chan<- prometheus.Metric) {
 		float64(numUsers),
 	)
 
+	seriesSpan.Finish()
+	flushQueueSpan := opentracing.StartSpan("flush_queue", opentracing.ChildOf(sp.Context()))
+
```

Comment:
Put the flush queue length stuff in a function and put the span creation in there too.